controller/options.go

```diff
@@ -41,6 +41,9 @@ type Options struct {
 
 	// Concurrency - The number of workers to use when processing the controller's workqueue.
 	Concurrency int
+
+	// PromoteFilterFunc filters the objects that are enqueued when the reconciler is promoted to leader.
```

Comment:
Can you add a comment on what returning true/false means in the context of the function and how it used?

Filter is always hard because true = the obj is filtered out? true = it passes the filter?

Reply:
Added more to the comment text, hopefully it's more clear now